packages/compiler/src/core/program.ts

```diff
@@ -244,18 +245,24 @@ export async function compile(
     program.reportDiagnostics(await linter.extendRuleSet(options.linterRuleSet));
   }
 
+  initializeSpinner("Checker...");
```

Comment:
I was more thinking you could do something like this
```
await action(() => asyncAction());
```
this allows the progress to report error if the promise fails too 
See in this pr in `init.ts` I did that for the install tsp dependency step https://github.com/microsoft/typespec/pull/6045/files